thorlcr/graph/thgraph.cpp

```diff
@@ -320,7 +320,46 @@ CGraphElementBase *createGraphElement(IPropertyTree &node, CGraphBase &owner, CG
     return container;
 }
 
-CGraphElementBase::CGraphElementBase(CGraphBase &_owner, IPropertyTree &_xgmml) : owner(&_owner)
+///////////////////////////////////
+static const StatisticsMapping statsMap({});
+CActivityCodeContext::CActivityCodeContext() : stats(statsMap)
```

Comment:
I expected this to be using the activity's stats.  Was there a reason why you created a new stats object instead?

Reply:
@ghalliday - none of the activity stats will be recorded in this context.
This (ICodeContext) is only used by generated code when registering function timings, so all stats will be nested.

These nested stats are then merged into the activity stats during collection, inside CSlaveActivity::serializeStats

Reply:
@jakesmith The alternative I was expecting (similar to roxie) is to pass the activity stats from the activity as a reference.  They wouldn't then need to be merged.

Reply:
right agree that would be better, though it's a little awkward because of the existing framework.
In that the activity context is created in a generic container before the specific activity with the stats. is created.

But I've changed it so that it is passed through when the activity is created, which avoids the merge (see latest commit).